### PR TITLE
[NEW] Send notification when login

### PR DESCRIPTION
**Pull request by albuquerquefabio**
## Proposed changes (including videos or screenshots)

image

### CODE REVIEW

File: apps/meteor/server/startup/migrations/v271-deviceManagement.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    }
+    .rc-color {
+      color: #F5455C;
+    }`;
+		await Settings.updateOne(
+			{
+				_id: 'email_style',
+			},
+			{
+				$set: {
```
Comment (sampaiodiego) [inline apps/meteor/server/startup/migrations/v271-deviceManagement.ts:110]: you should not overwrite the whole setting like this, doing so will discard/revert any change the sys admin might have done.

I saw the only change to the setting was a new `.rc-color` class.. so for this migration I'd suggest you to just add the new class to the existing value, this way no change will be lost during the migration.

File: apps/meteor/server/startup/migrations/v271-deviceManagement.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import { addMigration } from '../../lib/migrations';
+import { Settings } from '../../../app/models/server/raw';
+
+addMigration({
+	version: 267,
```
Comment (sampaiodiego) [inline apps/meteor/server/startup/migrations/v271-deviceManagement.ts:5]: the version is incorrect.. it should be `271` per file name.

File: apps/meteor/ee/server/lib/deviceManagement/session.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+		const user = await Users.findOneById<IUser>(userId, { projection: { name: 1, username: 1, emails: 1 } });
+		if (user?.emails?.length && !connection.loginToken) {
+			const {
+				name,
+				username,
+				emails: [{ address: email }],
+			} = user;
+			const { browser, os, device, cpu, app } = await uaParser(connection.httpHeaders['user-agent']);
+
+			let mailData = {};
```
Comment (sampaiodiego) [inline apps/meteor/ee/server/lib/deviceManagement/session.ts:51]: I saw a lot of duplicated code when defining a new value for `mailData` in the `switch` below, instead I'd recommend to change to a `const` and provide the common values here and just provide the new value on the switch statement, something like this:
```suggestion
			const mailData = {
				name,
				username,
				ipInfo: connection.clientAddress,
			};
```

File: apps/meteor/ee/server/lib/deviceManagement/session.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	if (uaString && UAParserDesktop.isDesktopApp(uaString)) {
+		rcAgent = UAParserDesktop.uaObject(uaString);
+	}
+
+	const ua = new UAParser(uaString);
+	return { ...ua.getResult(), ...rcAgent };
+};
+
+export const listenSessionLogin = async (): Promise<void> => {
+	sauEvents.on('accounts.login', async ({ userId, connection }) => {
```
Comment (sampaiodiego) [inline apps/meteor/ee/server/lib/deviceManagement/session.ts:39]: even though this works, you're adding a "hidden" dependency to the SAU Monitor service, which is not ideal.

what you should do instead, is to create a new service (let's say a `DeviceManagementService`) and add a listener to the original `accounts.login` event, just like the SAU Monitor service does, as you can see below:

https://github.com/RocketChat/Rocket.Chat/blob/33f28392697fcf93eceb23982e3e9451799e93e8/apps/meteor/server/services/sauMonitor/service.ts#L13

the difference though is that it should be an EE service, pretty much like `LDAPEEService`

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: apps/meteor/server/startup/migrations/v271-deviceManagement.ts (hunk 1 above)
Comment (albuquerquefabio) [reply apps/meteor/server/startup/migrations/v271-deviceManagement.ts:110]: I added `apps/meteor/app/lib/server/startup/email.ts` where the content is also created.

**Comment by sampaiodiego** [comment]
already merged into #25791

**Pull request closed without merging**